gobblin-metastore/src/main/java/org/apache/gobblin/metastore/MysqlJobStatusStateStore.java

```diff
@@ -33,7 +33,7 @@
  *
  * @param <T> state object type
  **/
-public class MysqlJobStatusStateStore<T extends State> extends MysqlStateStore {
+public class MysqlJobStatusStateStore<T extends State> extends MysqlStateStore implements DatasetStateStore {
```

Comment:
Can't MySqlJobStatusStateStore extend MysqlDatasetStateStore directly?

Reply:
Not easily.
MysqlDatasetStateStore implements DatasetStateStore<JobState.DatasetState>
we want MysqlDatasetStateStore to implement DatasetStateStore<State>